sdks/v4-sdk/src/PositionManager.ts

```diff
@@ -252,8 +257,17 @@ export abstract class V4PositionManager {
       planner.addIncrease(options.tokenId, position.liquidity, amount0Max, amount1Max, options.hookData)
     }
 
-    // need to settle both currencies when minting / adding liquidity
-    planner.addSettlePair(position.pool.currency0, position.pool.currency1)
+    // If migrating, we need to settle and sweep both currencies individually
+    if (isMint(options) && options.migrate) {
+      // payer is v4 positiion manager
+      planner.addSettle(position.pool.currency0, false)
+      planner.addSettle(position.pool.currency1, false)
+      planner.addSweep(position.pool.currency0, MSG_SENDER)
```

Comment:
sweep to recipient instead of msg sender? ie in migrator case msg.sender will be UR

Reply:
yesss